target/linux/ath79/dts/qca9558_tl-wr1043nd-v2.dts

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	compatible = "tplink,tl-wr1043nd-v2", "qca,qca9557";
+	model = "TP-Link TL-WR1043ND Version 2";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-output-names = "ref";
+		clock-frequency = <40000000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		system {
+			label = "tp-link:green:system";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "heartbeat";
+		};
+
+		usb {
+			label = "tp-link:green:usb";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		wifi_green {
+			label = "tp-link:green:wlan";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		wifi_wps {
+			label = "tp-link:green:wps";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <100>;
+
+		button@0 {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		button@1 {
+			label = "RFKILL button";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+//&pcie0 {
+//	status = "okay";
+//};
+
+&uart {
+	status = "okay";
+};
+
+&pll {
+	clocks = <&extosc>;
+};
+
+&spi {
+	status = "okay";
+	num-cs = <1>;
+
+	flash@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "m25p80";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x020000>;
+				read-only;
+			};
+
+			partition@1 {
```

Comment:
your partition table looks bogus. It rather should be something like:

```
		partitions {
			compatible = "fixed-partitions";
			#address-cells = <1>;
			#size-cells = <1>;

			partition@0 {
				label = "u-boot";
				reg = <0x000000 0x020000>;
				read-only;
			};

			partition@20000 {
				label = "firmware";
				reg = <0x020000 0x7d0000>;
			};

			partition@7f0000 {
				label = "art";
				reg = <0x7f0000 0x010000>;
				read-only;
			};
		};
```

The split of the firmware partition into kernel, rootfs and rootfs_data is done by the OpenWrt tplink firmware splitter.

Reply:
The mtd splitter isn't enabled in the kernel config. `CONFIG_MTD_SPLIT_TPLINK_FW=y` is missing.

Reply:
that worked, thank you !
do you have some hints on how to read the MAC address ?

Reply:
TP-Link devices often have mac in in u-boot mtd part and other devices have it in art or config...
You have to find the offset from the beginning of mtd part and define it under &ethX as shown below...

u-boot: partition@0 {
    label = "u-boot";
    reg = <0x000000 0x020000>;
    read-only;
};
art: partition@7f0000 {
    label = "art";
    reg = <0x7f0000 0x010000>;
    read-only;
};
...
&ethX {
...
    mtd-mac-address = <&u-boot 0xXXXX>; (or) mtd-mac-address = <&art 0xXXXX>;
...
}